### ISSUE TITLE
load dataset (load_tox21) with removed NaN values of targets

### ISSUE THREAD

**Issue by shenwanxiang** [deepchem/deepchem#1696]
**Description:**
import deepchem as dc
_, data, _ = dc.molnet.load_tox21(featurizer = 'Raw', split='random')
train, test, valid = data
train.y has fill NaN values with zeros

- DeepChem Version:
- Platform:

```
Your code sample here
```

**Comment by peastman** [comment]
Why would values be NaN?  I just checked the raw data file and it contains no NaNs in it.

If you're asking about how to recognize missing datapoints, they're marked by having the corresponding element of the weight matrix set to 0.

**Comment by shenwanxiang** [comment]
Hi Peastman, 
Thanks for clarifying! but what do you mean of " the weight matrix" ? 
if i read the csv file(tox21 dataset: https://github.com/deepchem/deepchem/blob/master/datasets/tox21.csv.gz) directly, then i will mask the missing label as "-1" instead of "0", because the origin 12 labels values are "0" or "1".

Besides, the tox21 dataset contains 7831 compounds, however, in the paper of  "MoleculeNet: a benchmark for molecular machine learning"(https://pubs.rsc.org/en/content/articlepdf/2018/sc/c7sc02664a) , they said this dataset contains 8014 compounds.

Is that the author didn't use this dataset as a benchmark in their paper? Thanks!

**Comment by peastman** [comment]
> but what do you mean of " the weight matrix" ? if that the values of train.w ?

Correct.  When computing the loss function, each term gets multiplied by the corresponding element of that matrix.  If it's set to 0 for a particle sample/task, then that value doesn't contribute to the loss.

**Comment by shenwanxiang** [comment]
> > but what do you mean of " the weight matrix" ? if that the values of train.w ?
> 
> Correct. When computing the loss function, each term gets multiplied by the corresponding element of that matrix. If it's set to 0 for a particle sample/task, then that value doesn't contribute to the loss.

Thanks!

**Comment by rbharath** [comment]
I'm going to close this old issue. Please feel free to re-open if you're still having issues!
